File: Programmers/[81303] 표 편집/woong-jae/81303.js
class ListNode {
    constructor(n, prev = null, next = null) {
        this.value = n;
        this.prev = prev;
        this.next = next;
    }
}

function solution(n, k, cmd) {
    let head = null, tail = null;
    for(let i = 0; i < n; i++) {
        if(!head) {
            head = new ListNode(i);
            tail = head;
        }
        else {
            tail.next = new ListNode(i, tail);
            tail = tail.next;
        }
    }
    
    const stack = [];
    let cur = head;
    while(k--) cur = cur.next;
    cmd.forEach(command => {
        let [dir, x] = command.split(" ");
        if(dir === "C") {
            stack.push(cur);
            if(cur.prev) cur.prev.next = cur.next;
            if(cur.next) {
                cur.next.prev = cur.prev;
                cur = cur.next;
            }
            else cur = cur.prev;
        }
        else if(dir === "Z") {
            const restore = stack.pop();
            if(restore.next) restore.next.prev = restore;
            if(restore.prev) restore.prev.next = restore;
        }
        else {
            if(dir === "U") while(x--) cur = cur.prev;
            else while(x--) cur = cur.next;
        }
    });
    
    let answer = Array(n).fill("O");
    stack.forEach(({value}) => {
       answer[value] = "X";
    });
    
    return answer.join("");
}

File: Programmers/[81303] 표 편집/Johoseong/81303.py
def solution(n, k, cmd):
    answer = ['O' for i in range(1, n +1)]
    linked_list = { i: [i - 1, i + 1] for i in range(1, n + 1)}
    pointer = k + 1
    stack = []

    for c in cmd:
        if c[0] == 'U':
            for _ in range(int(c[2:])): # 인덱스 2 이후까지 숫자 있는거 생각하기
                pointer = linked_list[pointer][0]
        elif c[0] == 'D':
            for _ in range(int(c[2:])):
                pointer = linked_list[pointer][1]
        elif c[0] == 'C':
            answer[pointer - 1] = 'X'
            prev, next = linked_list[pointer]
            stack.append([pointer, prev, next])

            if next == n + 1: # 표 마지막 행 삭제했으면 그 전 값 가르킴
                pointer = linked_list[pointer][0]
            else: # 디폴트는 삭제한 행 다음 행 가르킴
                pointer = linked_list[pointer][1]
            
            if prev == 0:
                linked_list[next][0] = prev
            elif next == n + 1:
                linked_list[prev][1] = next
            else:
                linked_list[prev][1] = next
                linked_list[next][0] = prev
                
        else:
            now, prev, next = stack.pop()
            answer[now - 1] = 'O'

            if prev == 0:
                linked_list[next][0] = now
            elif next == n + 1:
                linked_list[prev][1] = now
            else:
                linked_list[prev][1] = now
                linked_list[next][0] = now

    return ''.join(answer)